packages/ember-glimmer/lib/renderer.js

```diff
@@ -40,6 +40,63 @@ class DynamicScope {
   }
 }
 
+let nextRootId = 0;
+class RootState {
+  constructor(env, root, template, self, parentElement, dynamicScope) {
+    assert(`You cannot render \`${self.value()}\` without a template.`, template);
+
+    this.id = ++nextRootId;
```

Comment:
Don't we already have ids on views and a registry that we could use to weakly associate meta?